aim/common/utils.py

```diff
@@ -139,13 +155,15 @@ def snake_to_lower_camel(name):
 
 
 def sanitize_name(type, *args):
+    # Updated due to PY3 error
+    # TypeError: Unicode-objects must be encoded before hashing
     h = hashlib.sha256()
-    h.update(type)
-    h.update('\x00')
+    h.update(type.encode('utf-8'))
+    h.update('\x00'.encode('utf-8'))
     for component in args:
-        h.update(component)
-        h.update('\x00')
-    return base64.b32encode(h.digest()).rstrip('=').lower()
+        h.update(component.encode('utf-8'))
+        h.update('\x00'.encode('utf-8'))
+    return (base64.b32encode(h.digest()).decode('utf-8').rstrip('=').lower())
```

Comment:
base64 here should be the one "from oslo_serialization import base64" according to this guide,
https://wiki.openstack.org/wiki/Python3#Serialization:_base64.2C_JSON.2C_etc.
So you will need to at least update the import then see.

Reply:
But in oslo_serialization, there is no base32 library (note: b32encode() encodes it into base32 format, which is included in native `import base64` )
So I think we need to stick to native 'base64' library for this.

Reply:
We need to ensure that any names return the same results, so as long as it does that, we're good. Fortunately it looks like some UTs were created to cover this:
https://github.com/noironetworks/aci-integration-module/blob/master/aim/tests/unit/test_utils.py#L74-L89

I think if it passes the UTs, that should be good enough.